docs/rfc/active/graph_ql_frontend/README.md

```diff
@@ -0,0 +1,607 @@
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                return DaoFactory.getDatasetsDao().getDatasets(keys);
+            } catch (Exception e) {
+            throw new RuntimeException("Failed to batch load Datasets", e);
+            }
+        });
+        }
+    };
+DataLoader datasetLoader = DataLoader.newDataLoader(datasetBatchLoader);
+
+// Create CorpUser Loader
+BatchLoader<String, com.linkedin.identity.CorpUser> corpUserBatchLoader = new BatchLoader<String, com.linkedin.identity.CorpUser>() {
+    @Override
+    public CompletionStage<List<com.linkedin.identity.CorpUser>> load(List<String> keys) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                return DaoFactory.getCorpUsersDao().getCorpUsers(keys);
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to batch load CorpUsers", e);
+            }
+        });
+    }
+};
+DataLoader corpUserLoader = DataLoader.newDataLoader(corpUserBatchLoader);
+```
+In extending `BatchLoader`, a single batch "load" method must be provided. This API is exploited by the GraphQL engine, which aggregates calls for entities of the same type, reducing the number of downstream calls made within a single query. 
+
+*Registering a DataLoader* 
+
+DataLoaders are registered with the `DataLoaderRegistry` and subsequently included as input to the GraphQL engine:
+
+```java
+/*
+ * Build DataLoader Registry
+ */
+DataLoaderRegistry registry = new DataLoaderRegistry();
+registry.register("datasetLoader", datasetLoader);
+registry.register("corpUserLoader", corpUserLoader);
+
+/*
+ * Construct execution input
+ */
+ExecutionInput executionInput = ExecutionInput.newExecutionInput()
+.query(queryJson.asText())
+.variables(variables)
+.dataLoaderRegistry(register)
+.build();
+
+/*
+ * Execute GraphQL Query
+ */
+ExecutionResult executionResult = _engine.execute(executionInput);
+```
+
+For more information about `DataLoaders` see the [Using DataLoader](https://www.graphql-java.com/documentation/v15/batching/) doc. 
+
+For a full reference implementation of the query execution process, see the  ``GraphController.java`` class associated with this PR. 
+
+
+### Onboarding GMA Models
+
+The first phase of the GQL rollout will support primary-key lookups of GMA entities and projection of their associated aspects. In order to achieve this, we will
+- model entities, aspects, and the relationships among them
+- expose queries against top-level entities
+using GQL. 
+
+The proposed steps for onboarding a GMA entity, it's related aspects, and the relationships among them will be outlined next. 
+
+####1. **Model an entity in GQL**
+
+The entity model should include its GMA aspects, as shown below, as simple object fields. The client will need not be intimately aware of the concept of "aspects". Instead, it should be concerned with entities and their relationships.
+
+*Modeling Dataset*
+```javascript
```

Comment:
I assume there's syntax highlight support for gql directly in GH?

Reply:
Let's find out :D